pkg/controller/monitor/monitor_controller.go

```diff
@@ -317,6 +376,45 @@ func (r *ReconcileMonitor) Reconcile(ctx context.Context, request reconcile.Requ
 	return reconcile.Result{}, nil
 }
 
+// getCertificateBundle creates a configmap with a bundle for mTLS with other components.
+func getCertificateBundle(ctx context.Context, cli client.Client, cm *operatorv1.CertificateManagement) (*corev1.ConfigMap, error) {
+	pem := strings.Builder{}
+	addPem := func(secretName string) error {
```

Comment:
I don't believe you need an anonymous function here, could you please either put the contents of this inline in the loop below or make a separate non anonymous function for this (and please evaluate the errors in the loop).